src/pages/app/lib/context.tsx

```diff
@@ -1 +1,70 @@
+import React from 'react';
+import {
+  type BackgroundToAppMessage,
+  BACKGROUND_TO_APP_CONNECTION_NAME as CONNECTION_NAME,
+  MessageManager,
+  type AppToBackgroundMessage,
+} from '@/shared/messages';
+import { useBrowser } from '@/pages/shared/lib/context';
+import { dispatch } from './store';
+
 export { useBrowser, useTranslation } from '@/pages/shared/lib/context';
+
+export function WaitForStateLoad({ children }: React.PropsWithChildren) {
+  const message = useMessage();
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  React.useEffect(() => {
+    async function get() {
+      const response = await message.send('GET_CONTEXT_DATA');
```

Comment:
I think this should be a separate message, so app gets its own data, which could be different than popup's data.